Move all external images to within the repo

akleemans opened the pull request:
Move all images from external sources to this repository.

Solves the following issue: https://github.com/RetroAchievements/docs/issues/158

Some notes:
* As discussed in the issue above (and noted in the contribution guidelines), image links to retroachievements.org are also considered external and are added to this repository.
* Links from `camo.githubusercontent.com` were not accessible directly anymore, so I used the images provided by archive.org ([Example](https://web.archive.org/web/20221021023644/https://camo.githubusercontent.com/0eb5659706615243b9b0a39f82df0cbc45edc5a5/68747470733a2f2f692e696d6775722e636f6d2f463543666f74742e706e67))
* Some images on translated pages have been updated to the newest in the English version. Examples: `memsizes.png` => `mem-inspector-size-diagram.png`, images on `memory-inspector.md` (`memoryinspector-num.png`, `memoryinspector-filter.png`, `memoryinspector-bookmark.png` etc.)
* Badge image names have been kept according to the RA site ID.
* The devquest images have been numbered accordingly to the devquest ID.
* Image names are now all consistently using a dash (`-`), some underscores (`_`) were replaced.
* I added an `alt` text to the images where missing, and tried to replace generic `![image](...)` descriptions with something more meaningful. I did not fix all existing `alt` texts, they are not all aligned currently, but I focussed on the image replacements.
* I didn't replace the labels in `docs/guidelines/content/hash-labels.md` (yet) - I was hesitant if those should be duplicated to this repo, as they are referenced directly (and used as image tags) from all the "Supported Game Files" pages. If they should also be included here please let me know. (I did update the image links in the `pt` version, there was an update to the image links in the English version which was not applied in `pt`).
* If there were more than to 5-10 images for a single topic, I tried to prefix the images, like `rabot-*.png` for the Discord RA Bot page. If I get it correctly a flat directory structure of `public` is desirable, so I didn't create subfolders. 
* Checked for duplicate images using image checksums


❓Question: When running the docs locally, I got warnings:
> Files in the public directory are served at the root path.
> Instead of /public/badge-64969.png, use /badge-64969.png.

But the `contributing-to-the-docs.html` explicitly states to use image links with the `/public` prefix:
> ✅ GOOD
> ...(/public/current-branch.png)

Can the local warnings be ignored?

akleemans committed:
Replace bit.ly URLs

akleemans committed:
Adapt pt/ docs to changes in main hash-labels.md

akleemans committed:
Replace links to badges with local images

akleemans committed:
Add mario_time.png

akleemans committed:
Rename badges

akleemans committed:
Add DevQuests, update badge casing

akleemans committed:
Replace imgur links

akleemans committed:
Update image descriptions

akleemans committed:
Add some github-userimages, add image descriptions

akleemans committed:
Move some more images from githubusercontent, fix "Let's" typo

akleemans committed:
Refactor image names to consistent use of "-"

akleemans committed:
Clean up devquest image names

akleemans committed:
Add more images from githubusercontent

akleemans committed:
Replace rest of githubusercontent images

akleemans committed:
Fix image links

akleemans committed:
Revert typo fixes, focus on image replacement

akleemans committed:
Update RP casing for pt version

akleemans committed:
Fix image link for mario-time

akleemans committed:
Fix image link (...-png)

akleemans committed:
Add missing badge

akleemans committed:
Fix image link to reset_password.png

akleemans committed:
Clean up echo-ram duplicate

akleemans committed:
Remove remaining duplicate images

wescopeland commented:
> When running the docs locally, I got warnings

Good question. It's possible VitePress changed something in a recent update.

I think we should probably:
* Respect the warning and update the paths.
* Update our own contributing docs to correct the error you've pointed out.

akleemans commented:
@wescopeland So should I update the paths in this PR, or do we do this later? The change you suggest will also affect all other paths of existing `/public/` images which I didn't touch here.

 And another question, how does the PR review process work, do I have to ping somewhere or will it get picked up eventually by somebody?

akleemans commented:
@wescopeland How should we continue here? I can also make multiple smaller PRs if this one is too much at once.
Or if the issue mentioned is not relevant anymore please let me know so I can cancel the PR. Thanks!

wescopeland commented:
Hi @akleemans! Apologies for the delay here.

> should I update the paths in this PR, or do we do this later?

I think we should respect the warning and update the paths, even though it's slightly larger in scope than originally planned.

akleemans committed:
Update image links by removing /public

akleemans committed:
Update contributing-to-the-docs.md and add explanation for image links

akleemans commented:
@wescopeland Updated the PR.
* Updated all image links by removing the `/public` part. No warnings appear now anymore when I run it with VitePress.
* Updated the contribution docs (example and explanation)

akleemans commented:
@wescopeland From my point of view this PR would be ready for review, please let me know if there's anything missing.

wescopeland commented:
Hi! Thanks for your patience. I'll do my best to get a review in tomorrow.

wescopeland reviewed (approved):
Verified locally. Thank you!
